src/Fabulous.Avalonia/Widgets/Controls/NumericUpDown.fs

```diff
@@ -0,0 +1,176 @@
+﻿namespace Fabulous.Avalonia
+
+open System.Runtime.CompilerServices
+open Avalonia.Controls
+open Fabulous
+open System
+open Avalonia.Interactivity
+
+type IFabNumericUpDown =
+    inherit IFabTemplatedControl
+
+module NumericUpDown =
+    let WidgetKey = Widgets.register<NumericUpDown> ()
+
+    let AllowSpin =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.AllowSpinProperty
+
+    let ButtonSpinnerLocation =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.ButtonSpinnerLocationProperty
+
+    let ClipValueToMinMax =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.ClipValueToMinMaxProperty
+
+    let FormatString =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.FormatStringProperty
+
+    let HorizontalContentAlignment =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.HorizontalContentAlignmentProperty
+
+    let VerticalContentAlignment =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.VerticalContentAlignmentProperty
+
+    let Increment =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.IncrementProperty
+
+    let IsReadOnly =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.IsReadOnlyProperty
+
+    let Maximum =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.MaximumProperty
+
+    let Minimum =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.MinimumProperty
+
+    let NumberFormat =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.NumberFormatProperty
+
+    let ParsingNumberStyle =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.ParsingNumberStyleProperty
+
+    let ShowButtonSpinner =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.ShowButtonSpinnerProperty
+
+    let Text = Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.TextProperty
+
+    let TextConverter =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.TextConverterProperty
+
+    let Watermark =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.WatermarkProperty
+
+    let Value =
+        Attributes.defineAvaloniaPropertyWithEquality NumericUpDown.ValueProperty
+
+    let ValueChanged =
```

Comment:
See my previous comment. For this one, we can use `Attributes.defineAvaloniaPropertyWithChangedEvent` with converters to massage the value :) 
```fsharp
let ValueChanged =
        Attributes.defineAvaloniaPropertyWithChangedEvent
            "NumericUpDown_ValueChanged"
            NumericUpDown.ValueProperty
            Option.toNullable
            Option.ofNullable
```